snuba/snuba_migrations/events_analytics_platform/0050_add_attributes_array_column.py

```diff
@@ -0,0 +1,102 @@
+from snuba.clickhouse.columns import JSON, Column
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget
+
+storage_set = StorageSetKey.EVENTS_ANALYTICS_PLATFORM
+table_name_prefix = "eap_items_1"
+new_columns = [
+    Column(
+        "attributes_array",
+        JSON(
+            max_dynamic_paths=4096,
```

Comment:
this number is way too high. this would create an additional 4096 columns on each table, almost 14000 columns in total. and I'm fairly certain we will reach it. All it would take is 4096 differently named values across our entire customer base.

Reply:
I would recommend putting it quite low. like 32 and have the [shared structure](https://clickhouse.com/blog/json-data-type-gets-even-better#bucketed-shared-data) take care of the rest

Reply:
Default is 1024, ClickHouse is OK having a higher number of columns. I'm not sure we need to keep a very small number like 32.

We want to take advantage of the shared structure but we'll need to upgrade to 25.8 first. At least, we should keep the default value.